### ISSUE TITLE
Support for wrapped method

### ISSUE THREAD

**Issue by dingyaguang117** [mrk-andreev/tornado-swagger#14]
Methods maybe wrapped, could we add this ?

```
def _extract_parameters_names(handler, parameters_count):
    if parameters_count == 0:
        return []

    parameters = ['{?}' for _ in range(parameters_count)]

    for method in handler.SUPPORTED_METHODS:
        method_handler = getattr(handler, method.lower())

+       while hasattr(method_handler, '__wrapped__'):
+           method_handler = method_handler.__wrapped__

        args = inspect.getfullargspec(method_handler).args[1:]

        if len(args) > 0:
            for i, arg in enumerate(args):
                if set(arg) != {'_'}:
                    parameters[i] = arg

    return parameters

```

I changed the code above, and it works. but, I found you have modified this 8 months ago.

```
def _try_extract_docs(method_handler):
    try:
        return inspect.getfullargspec(method_handler).args[1:]
    except TypeError:  # unsupported callable
        if hasattr(method_handler, '__wrapped__'):
            return _try_extract_docs(method_handler.__wrapped__)
        else:
            return []


def _extract_parameters_names(handler, parameters_count):
    if parameters_count == 0:
        return []

    parameters = ['{?}' for _ in range(parameters_count)]

    for method in handler.SUPPORTED_METHODS:
        method_handler = getattr(handler, method.lower())
        args = _try_extract_docs(method_handler)

        if len(args) > 0:
            for i, arg in enumerate(args):
                if set(arg) != {'_'}:
                    parameters[i] = arg

    return parameters
```

will you release a new  version?

**Comment by dingyaguang117** [comment]
I think 

```
def _try_extract_docs(method_handler):
    try:
        return inspect.getfullargspec(method_handler).args[1:]
    except TypeError:  # unsupported callable
        if hasattr(method_handler, '__wrapped__'):
            return _try_extract_docs(method_handler.__wrapped__)
        else:
            return []
```
can't solve this problem

My decorator below can't be parsed correctly.
```
def valid_token(allow_guest=False):
    def decorator(method):
        @functools.wraps(method)
        def wrapper(self, *args, **kwargs):
            # do some thing
            return method(self, *args, **kwargs)
        return wrapper
    return decorator
```

**Comment by mrk-andreev** [comment]
This bug was fixed in release v1.2.2. 

If you want to get original function (remove wrappers), you can use `inspect.unwrap` (https://github.com/mrk-andreev/tornado-swagger/blob/master/tornado_swagger/_builders.py#L45) instead `while + __wrapped__`.

**Comment by libinghope** [comment]
If the method is wrapped, docs  can't be generated
why???
my decorator likes this below
> def has_permission(permission):
>     def check_permission(method):
>         def wrapper(self, *args, **kwargs):
>             if not (self.current_user and self.current_user.has_key("role")):
>                 raise tornado.web.HTTPError(403)
>            .......
>             return method(self, *args, **kwargs)
> 
>         return wrapper
>     return check_permission
